src/recording/macromanager.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <QObject>
+
+#include "engine/enginemaster.h"
+#include "mixer/playermanager.h"
+#include "preferences/usersettings.h"
+#include "recording/recordingmanagerbase.h"
+
+#define MAX_MACRO_SIZE 1000
+#define NO_DECK nullptr
+
+struct MacroAction {
+    MacroAction(){};
+    MacroAction(double position, double target)
+            : position(position), target(target){};
+    double position;
+    double target;
+};
+
+struct Macro {
+    int length = 0;
+    MacroAction actions[MAX_MACRO_SIZE];
+    QString deck = NO_DECK;
```

Comment:
QString is not a pointer!  Why not https://doc.qt.io/qt-5/qstring.html#clear

Reply:
Ah, didn't know. It worked :P